### ISSUE TITLE
panicked at 'Attempted to get child of leaf'

### ISSUE THREAD

**Issue by eadf** [dbyr/rb_tree#3]
Hi! 

I just got a strange error from RBMap 0.3.2.  OS X 11.2 
I've tried to distill the problem down to a bare minimum, and this is what triggers it.

```rust
#[test]
fn rbmap_test() {
    let mut rbmap = rb_tree::RBMap::<usize, usize>::new();

    rbmap.insert(17, 0);
    rbmap.insert(18, 0);
    rbmap.insert(19, 0);
    rbmap.insert(20, 0);
    rbmap.insert(21, 0);
    rbmap.insert(22, 0);
    rbmap.insert(23, 0);
    rbmap.insert(24, 0);
    rbmap.insert(25, 0);
    rbmap.insert(26, 0);
    rbmap.insert(27, 0);
    rbmap.insert(28, 0);
    let k=27;
    rbmap.remove(&k);
    let k=26;
    rbmap.remove(&k);
}
```

thread 'rbmap_test' panicked at 'Attempted to get child of leaf', ...../rb_tree-0.3.2/src/node.rs:246:24

cargo 1.51.0-nightly (783bc43c6 2021-01-20)
cargo 1.49.0 (d00d64df9 2020-12-05)
cargo 1.51.0-nightly (ab64d1393 2021-02-10)
cargo 1.50.0 (f04e7fab7 2021-02-04)

all gives the same result

**Comment by dbyr** [comment]
Thanks for providing the example, it gives me the same result. I'll take a look at it!

**Comment by dbyr** [comment]
Fixed the issue in https://github.com/dbyr/rb_tree/commit/59153460476ea6173b56438b9b5596f25a9c1ddb and also added your example to the tests as a regression. Updated version available in v0.3.3. Thanks for letting me know!
